### ISSUE TITLE
Gradient removed if used by two rects

### ISSUE THREAD

**Issue by codegain** [svg/svgo#1282]
Hi,

I have the following SVG:

```
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 40 40">
<defs>
<style>.cls-1{fill:url(#a);}</style>
<linearGradient id="a" x1="20" y1="0.08" x2="20" y2="39.95" gradientUnits="userSpaceOnUse">
    <stop offset="0" stop-color="#ff0000"/>
    <stop offset="1" stop-color="#0000ff"/>
</linearGradient>
</defs>
<rect class="cls-1" x="0" y="0" width="15" height="40" />
<rect class="cls-1" x="20" y="0" width="15" height="40"/>
</svg>
```

With these options for SVGO: (I use some JS wrapper)

```
        plugins: [
            {
                removeScriptElement: true
            },
            {
                removeUnknownsAndDefaults: {
                    keepDataAttrs: false
                },
            },
            {
                cleanupIDs: {
                    force: true
                }
            },
        ]
```

the resulting SVG looks like this:

```
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 40 40"><path d="M0 0h15v40H0zm20 0h15v40H20z" fill="url(#a)"/></svg>
```

which does not contain the gradient in the above SVG.

If I add this to my options:

```
            {
                inlineStyles: {
                    onlyMatchedOnce: false
                }
            }
```

the resulting SVG is:

```
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 40 40"><defs><linearGradient id="a" x1="20" y1=".08" x2="20" y2="39.95" gradientUnits="userSpaceOnUse"><stop offset="0" stop-color="red"/><stop offset="1" stop-color="#00f"/></linearGradient></defs><path fill="url(#a)" d="M0 0h15v40H0zm20 0h15v40H20z"/></svg>
```

which is correct. So why does the above example simply remove the `linearGradient` but still references it? This seems like a bug to me.

**Comment by codegain** [comment]
I should add, that if I use only one rect:

```
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 40 40">
<defs>
<style>.cls-1{fill:url(#a);}</style>
<linearGradient id="a" x1="20" y1="0.08" x2="20" y2="39.95" gradientUnits="userSpaceOnUse">
    <stop offset="0" stop-color="#ff0000"/>
    <stop offset="1" stop-color="#0000ff"/>
</linearGradient>
</defs>
<rect class="cls-1" x="0" y="0" width="15" height="40" />
</svg>
```

and dont use `inlineStyles/onlyMatchedOnce: false`, the resulting SVG will be:

```
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 40 40"><defs><linearGradient id="a" x1="20" y1=".08" x2="20" y2="39.95" gradientUnits="userSpaceOnUse"><stop offset="0" stop-color="red"/><stop offset="1" stop-color="#00f"/></linearGradient></defs><path fill="url(#a)" d="M0 0h15v40H0z"/></svg>
```

This also is correct as this does not remove the gradient.

**Comment by maheshwaran-p** [comment]
I need only inline styled svg after svgopreprocess can anyone help me?

**Comment by maheshwaran-p** [comment]
Or any other solution to convert internal styled svg to inline styled svg ?

i found similar platform in python :
https://github.com/scour-project/scour

**Comment by KTibow** [comment]
I can't reproduce this on the latest version (tried a variety of configs), this probably can be closed

**Comment by codegain** [comment]
@KTibow Yes this is fixed with the latest version (3.1.0). I was using 2.8.0 at that time.
